src/Templating/Helper/NumberHelper.php

```diff
@@ -345,4 +342,21 @@ protected function parseConstantValue($attribute)
 
         return \constant($constantName);
     }
+
+    private function parseNumericValue($number)
+    {
+        if (!\is_scalar($number)) {
+            throw new \TypeError('Number must be a scalar value');
+        }
+
+        if (\is_string($number)) {
+            if (!\is_numeric($number)) {
+                throw new \typeError('Number must be a numeric string')
+            }
+
+            $number = (float) $number;
+        }
+
+        return $number;
```

Comment:
```
if (\is_float($number) || \is_int($number)) {
     return $number;
}
if(\is_numeric($number)) {
     return (float) $number;
}
throw \TypeError(...)
```

`\is_scalar` returns `true` for boolean. And only one TypeError message is enough.

Reply:
done